src/App.js

```diff
@@ -28,31 +28,100 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
-
+  const [isXTurn, setXTurn] = useState(true);
+  const [winner, setWinner] = useState(null);
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+ 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+
+  const onClickCallback = ( (squareId) => {
+
+    // will not continue the game if there is already a winner
+    if (winner) return;
+
+    let copiedSquares = [...squares];
+    for(let row of copiedSquares) {
+      for(let square of row) {
+       if (square.id === squareId && square.value === '') {
```

Comment:
This is one of a few lines where your indentation is just slightly off. I use a VSCode plugin called indent-rainbow to help me catch things like this.